EIPS/eip-6963.md

```diff
@@ -234,109 +230,32 @@ function onPageLoad(): EIP6963ProviderDetail[] => {
+
+### Wallet Imitation and Manipulation
 
-The use of SVG images introduces a cross-site scripting risk as they can include JavaScript code. Applications and libraries must render SVG images using the `<img>` tag to ensure no JS executions can happen.
+Similarly so, DApp's are expected to actively detect for misbehavior of properties or functions being modified in order to tamper with or modify other wallets. One way this can be easily achieved is to look for when the `uuid` property within two `EIP6963ProviderInfo` objects match. DApps and DApp discovery libraries are expected to consider other potential methods that the `EIP6963ProviderInfo` objects are being tampered with and consider additional mitigation techniques to prevent this as well in order to protect the user.
 
-// TODO for @kdenhartog: Add a privacy section about Wallet fingerprinting and how Wallets should not announce themselves until the DApp has dispatched the request provider Event and gathered consent from the user that they wish to announce themselves to the DApp.
+### Prevent SVG Javascript Execution 
+
+The use of SVG images introduces a cross-site scripting risk as they can include JavaScript code. This Javascript executes within the context of the page and can therefore modify the page or the contents of the page. So when considering the experience of rendering the icons, DApps need to take into consideration how they'll approach handling these concerns in order to prevent an image being used as an obfuscation technique to hide malicious modifications to the page or to other wallets.
+
+### Prevent Wallet Fingerprinting
+
+One advantage to the concurrency Event loop utilized by this design is that it operates in a manner where either the DApp or the Wallet can initiate the flow to announce a provider. For this reason, Wallet implementers can now consider whether or not they wish to announce themselves to all pages or attempt alternative means in order to reduce the ability for a user to be fingerprinted by the injection of the `window.ethereum` object. Some examples, of alternative flows to consider would be to wait to inject the provider object until the DApp has announced the `eip6963:requestProvider`. At that point, the wallet can initiate a UI consent flow to ask the user if they would like to share their wallet address. This allows for the Wallet to enable the option of a "private connect" feature. However, if this approach is taken Wallet's must also consider how they intend to support backwards compatibility with a DApp that does not support this EIP.
```

Comment:
Sharing the *address* would in some cases imply a request of `eth_requestAccounts` method from the DApp right? In this case we are refering to the provider announcement.

I would think this more of a whitelist of trusted sites, so the Wallet doesn't need to ask everytime to announce its provider, although I know this is just an example n.n
```suggestion
One advantage to the concurrency Event loop utilized by this design is that it operates in a manner where either the DApp or the Wallet can initiate the flow to announce a provider. For this reason, Wallet implementers can now consider whether or not they wish to announce themselves to all pages or attempt alternative means in order to reduce the ability for a user to be fingerprinted by the injection of the `window.ethereum` object. Some examples of alternative flows to consider would be to wait to inject the provider object until the DApp has announced the `eip6963:requestProvider`. At that point, the wallet can initiate a UI consent flow to ask the user if they would like to share their wallet provider. This allows for the Wallet to enable the option of a "private connect" feature. However, if this approach is taken Wallet's must also consider how they intend to support backwards compatibility with a DApp that does not support this EIP.
```

Reply:
Not quite, by only sharing the wallet provider you'd double up the consent flows in an unnecessary way. Essentially what the UI is doing is eliminating the separate consent of injection and wallet address by presuming that if a user is willing to consent to sharing the address that they're fine to also inject the provider. Therefore, the only UI that needs to be initiated is to ask the user which wallet address they'd want to use, when the wallet detects the `requestProvider` Event, and at that point the wallet can both respond with the `announceProvider` and prepare itself to automatically respond to the `eth_requestAccounts` RPC call all in one go without needing a secondary UI consent flow.

Reply:
Maybe, this should be updated to say "... share their wallet address which implicitly grants consent to share the Wallet Provider."

Reply:
could be! I understand now! thanks for the explanation :)